src/datahike/migrate.clj

```diff
@@ -1,37 +1,62 @@
 (ns ^:no-doc datahike.migrate
   (:require [datahike.api :as api]
+            [datahike.datom :as d]
             [datahike.db :as db]
-            [clojure.java.io :as io]))
+            [clj-cbor.core :as cbor]))
 
 (defn export-db
   "Export the database in a flat-file of datoms at path."
   [db path]
-  (with-open [f (io/output-stream path)
-              w (io/writer f)]
-    (binding [*out* w]
-      (doseq [d (datahike.db/-datoms db :eavt [])]
-        (prn d)))))
+  (cbor/spit-all path (map seq (sort-by
+                                (juxt d/datom-tx :e)
+                                (api/datoms (if (db/-keep-history? db) (api/history db) db) :eavt)))))
 
-(defn update-max-tx-from-file
-  "Find bigest tx in file and update max-tx of db.
+(defn update-max-tx
+  "Find bigest tx in datoms and update max-tx of db.
   Note: the last tx might not be the biggest if the db
   has been imported before."
-  [db file]
-  (let [max-tx (->> (line-seq (io/reader file))
-                    (map read-string)
-                    (reduce #(max %1 (nth %2 3)) 0))]
-    (assoc db :max-tx max-tx)))
+  [db datoms]
+  (assoc db :max-tx (reduce #(max %1 (nth %2 3)) 0 datoms)))
+
+(defn- instance-to-date [v]
+  (if (instance? java.time.Instant v) (java.util.Date/from v) v))
 
 (defn import-db
   "Import a flat-file of datoms at path into your database."
   [conn path]
   (println "Preparing import of" path "in batches of 1000")
-  (swap! conn update-max-tx-from-file path)
-  (print "Importing ")
-  (time
-   (doseq  [datoms (->> (line-seq (io/reader path))
-                        (map read-string)
-                        (partition 1000 1000 nil))]
-     (print ".")
-     (api/transact conn (vec datoms)))))
+  (let [datoms (->> (cbor/slurp-all path)
+                    (map #(-> (apply d/datom %) (update :v instance-to-date))))]
+    (swap! conn update-max-tx datoms)
+    (print "Importing ")
+    (api/transact conn (vec datoms))))
```

Comment:
This means we are importing all datoms at once, so no partitions for now?

Reply:
I thought to drop them for now since with history, they introduce transaction datoms that don't belong to the imported database. They could be removed, but that seemed clunky, so I figured to drop partitions for now pending a better solution. But either approach makes sense—let me know.